airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -693,8 +717,23 @@ def request_params(
     ) -> MutableMapping[str, Any]:
         params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
         if not next_page_token:
-            params.update({"createdBefore": pendulum.now("utc").strftime(self.time_format)})
+            end_date = pendulum.now("utc").strftime(self.time_format)
+            if is_valid_replication_end_date(self._replication_end_date):
+                end_date = self._replication_end_date
+            params.update({self.replication_end_date_field: end_date})
         return params
 
     def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
         yield from response.json().get(self.data_field, {}).get("shippingLabels", [])
+
+
+def is_valid_replication_end_date(custom_date: str) -> bool:
```

Comment:
I don't think this function should exist, the regex pattern in the  `spec.json` should already guarantee a valid input.
If you still think this is required, please call this function once in the `__init__` of the stream class, avoid calling it multiple times.

Reply:
totally agree!!

have removed it